Utils/Dataflow/030_PDFAnalyzer/manager.py

```diff
@@ -1486,9 +1487,9 @@ def show_paper_datasets(self, window, paper):
                                             datasets)
         else:
             if not paper.datasets:
-                l = Tkinter.Label(window, text="No datasets found",
-                                  font=HEADING_FONT)
-                l.grid(row=0)
+                lbl = Tkinter.Label(window, text="No datasets found",
+                                    font=HEADING_FONT)
```

Comment:
This whitespace-only change should go in its own.

Reply:
@konvpalto,
not sure. Why not to fix indentation, broken due to the change of the variable name, in the same commit where the variable name was changed?
I mean, if it was a standalone codestyle fix -- you`re right. But if we try to follow the codestyle, it looks OK for me to make changes aimed **not to break** the codestyle along with the changes that provoke the anomaly... no?

The logic of "whitespace-only" commits is understood, but is it *really* applicable for this case?

Reply:
I agree with @mgolosova on this one. Whitespace changes could have been moved into a separate commit if, for example, the code was literally peppered with whitespace-related problems, so it would have been reasonable to fix other things while ignoring them, and then fix the whitespace in a separate commit. This is not the case here, here we pick a chunk of code with 1 problem, and transform it into a chunk of code with 0 problems. I see no reason to explicitly state in our commit history that it was achieved by performing not 1 but 2 changes, both of which are cosmetic and bear no real impact on program logic.

Same for other whitespace and linebreaking comments.

Reply:
My only wish there was to split the whitespace-only part into the second commit to allow for easier review: "look, ma, just whitespace stuff here".  It makes one to be sure what to check in the commit.

Besides: what's the commit message here? Bare "Fixed E741 - ambiguous variable names"?  Let me pretend to review this loudly:
 - Hmm, no mentions about whitespace changes?  Did they really wanted that?  Typo?
 - OK, let me try to study, may be that was an alignment problem (tab/space mix)?  No?
 - No, just whitespaces.  OK, probably, that what they also meant.

In reality, there are two questions.  The first one: "can this commit be improved?".  The answer is "yes" and I can bet for my life on that.  Because (taking "consistency" as the correspondence between various parts of an object and the fact that commit log and commit contents are *equally important* parts of the object /that is commit/) your commit is inconsistent, I am awfully sorry to say it.

And I see at least two ways to improve:
 - extend commit log;
 - split into two commits.
May be there are other ones.

The second question is "will those changes worth that/justify the time spent/etc?".  I dunno.  And, honestly, I don't care much: improvement for me is the justification on its own. For you -- you tell me.


And there is the third question, specifically to you, Vasily: why are you asking me your questions with "I agree with ..." stanza?  "Peppered" is a good and strong word, it, may I assume, provides the said justification: "oh, it's peppered, let's now unpepper that, it adds The Real Value (tm)".  What's the difference between one grain of pepper and (42-1) grans?  Just a loud thought, can be skipped.

But splitting commits is easy: so, what are you really trying to say or ask about?  Or, more explicitely: are you really trying to understand?  I'd just split it: no additional load on the repository, definitely, easier to review, easier to skim through in the future.  I used to answer such questions "Really Smart Boys can cope with such commits easily, whazzup?".  Yep, they do.  What's your idea here?  Can be moved to e-mail (or skipped: won't be rude by any means).


It is not a showstopper by itself, but, please, remember: the devil lies in the details. And, it seems to me, most of the time -- only in them.

---
And the last thing: it is just my opinion, but I don't currently see any arguments not to do that.  Everyone is free to skip this, do that, adopt or ask again.  The only thing that I'd forbid (if I could) is to make this a showstopper to the evolution (of the code).

Same for other comments from /me ;)

Reply:
@konvpalto 
I was stating my current position regarding this point and trying to explain my logic behind it, not asking questions.

>"Peppered" is a good and strong word, it, may I assume, provides the said justification: "oh, it's peppered, let's now unpepper that, it adds The Real Value (tm)". What's the difference between one grain of pepper and (42-1) grans? 

The difference in this case is amount of additional work created by task of removing pepper from somewhere. Removing one grain is pretty easy, while large amount of pepper may require the worker to get an empty sack to store removed pepper in it, and then dispose of the sack filled with pepper. You want the worker to get a sack for one grain as well. I don't insist on this comparison being 100% accurate/complete/whatever and using it to decide things in our problem, merely illustrating my point.

>Besides: what's the commit message here? Bare "Fixed E741 - ambiguous variable names"? Let me pretend to review this loudly:
>- Hmm, no mentions about whitespace changes? Did they really wanted that? Typo?
>- OK, let me try to study, may be that was an alignment problem (tab/space mix)? No?
>- No, just whitespaces. OK, probably, that what they also meant.

Ok, let me explain my logic as well. "Fixed E741, then fixed whitespace problem caused by doing so". Hmm, why haven't they told me how they powered up their PC before doing so? True, doing so does not affects the code (unlike your example), but it's pretty much the same to me, because i think that in most cases solutions for problems caused by fixes should be included in these fixes. Especially here, since the secondary problem does exactly the same thing as the main one - breaks the code style.

Reply:
> I was stating my current position regarding this point and trying to explain my logic behind it, not asking questions.

Then I am not interested in continuing this topic in its current state: I had already said everything I wanted to.  The additional work from splitting the commit into two is negligible, but if we're making two long comments trying to pursue (not get any new knowledge, just pursue) each other -- it doesn't worth it.  My point is that review-wise splitting is good (not for me, by the way: I am technically fine with commits that have more problems that this one); your point is that adding (helpful, but "obvious") information into the commit log is pointless.

My experience in digging up the code in various open-source repositories many times when it was needed or even been urgent tells me that you're wrong.  Perhaps you're doing more such digging than me and, at the same time, still find your point to be right; then, most probably, my technicall skills are miserable here.  If not -- you'll start to work with other's code, you'll learn.  Or not.

Dixi.

Reply:
And, regarding this splitting issue: everyone may pretend that it can be ignored and do that.